snuba/clickhouse/optimize/optimize.py

```diff
@@ -80,6 +89,11 @@ def run_optimize_cron_job(
     database = storage.get_cluster().get_database()
     optimize_scheduler = OptimizeScheduler(parallel=parallel)
 
+    # if theres a merge in progress wait for it to finish
+    while is_busy_merging(clickhouse, database, table):
+        logger.info(f"busy merging, sleeping for {OPTIMIZE_BASE_SLEEP_TIME}s")
+        time.sleep(OPTIMIZE_BASE_SLEEP_TIME)
```

Comment:
One optimization we could do is potentially add a higher bound on how much time to sleep. Usually merges take 1.5 hours to complete. So you could add an upper bound to protect from code logic bug. The upper bound could be 2 hours. So even if the signal says that there is an ongoing merge, you could proceed after 2 hours have passed from the check starting.

Reply:
That makes sense. Added the check to set upper bound to 2 hours.